### PR TITLE
Add active editor to editor tabs and use in variable resolver

### LINKED ISSUE #123028: Variable ${file} can not be resolved
Issue Type: <b>Bug</b>

I have a binary file that to convert I have made a task for.
For the task to run I have to "open" the file.
I do that, with the message coming up: "The file is not displayed bla bla bla...".
Somehow you changed former working behaviour in 1.56.
In 1.55 and before I could just run the task fine – now it is broken and I will get an error message saying:
`Variable ${file} can not be resolved. Please open an editor.`

Please revert this change so the file variable resolves properly again.

VS Code version: Code 1.56.0 (cfa2e218100323074ac1948c885448fdf4de2a7f, 2021-05-04T22:09:06.405Z)
OS version: Windows_NT x64 10.0.19042

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5700HQ CPU @ 2.70GHz (8 x 2694)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (3.95GB free)|
|Process Argv|--crash-reporter-id 6ad73eb2-e24b-4305-a8ef-19341314fe6c|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (48)</summary>

Extension|Author (truncated)|Version
---|---|---
calculate|aca|2.1.0
ifc-syntax|Ala|0.1.4
Bookmarks|ale|13.0.4
spellright|ban|3.0.56
change-color-format|bbu|1.0.0
gitignore|cod|0.6.0
aoe2-rms|del|0.3.3
rhinopython|des|0.1.11
dotfiles-syntax-highlighting|dot|0.1.0
vscode-gitignore-syntax|dun|0.1.2
EditorConfig|Edi|0.16.4
vscode-open-in-application|fab|1.0.6
auto-close-tag|for|0.5.10
gc-excelviewer|Gra|3.0.41
gdl|Gra|1.24.3
vscode-guid|hea|1.4.20
reg|ion|1.0.3
aoe2-aiscript|jvi|0.1.7
vscode-markdown-notes|kor|0.0.24
nim|kos|0.6.6
vscode-apache|mrm|1.2.0
vscode-duplicate|mrm|1.2.1
csharp|ms-|1.23.11
python|ms-|2021.4.765268190
vscode-pylance|ms-|2021.4.3
jupyter|ms-|2021.6.811652604
remote-ssh|ms-|0.65.4
remote-ssh-edit|ms-|0.65.4
remote-wsl|ms-|0.56.1
cpptools|ms-|1.3.1
hexeditor|ms-|1.6.0
powershell|ms-|2021.2.2
sublime-keybindings|ms-|4.0.9
resourcemonitor|mut|1.0.7
printcode|nob|3.0.0
material-icon-theme|PKi|4.6.0
polacode|pnp|0.3.4
run-on-save|puc|1.4.1
gruvbox-material|sai|6.4.1
sonarlint-vscode|Son|1.21.0
language-stylus|sys|1.11.0
vscode-open-in-github|sys|1.15.0
markdown-links|tch|0.8.0
open-in-browser|tec|2.0.0
shell-launcher|Tyr|0.4.1
sort-lines|Tyr|1.9.0
vscode-icons|vsc|11.4.0
vscode-todo-highlight|way|1.0.4

</details>

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #123028

### CODE REVIEW

File: src/vs/workbench/api/browser/mainThreadEditorTabs.ts [hunk 1]
```diff
@@ -69,7 +71,8 @@ export class MainThreadEditorTabs {
 				tabs.push({
 					group: group.id,
 					name: editor.getTitle(Verbosity.SHORT) ?? '',
-					resource: editor.resource
+					resource: editor.resource,
+					isActive: this._editorService.activeEditor?.resource?.toString() === editor.resource.toString()
```
Comment (jrieken) [inline src/vs/workbench/api/browser/mainThreadEditorTabs.ts:75]: I believe you can also figure this out via `IEditorGroupsService#activeGroup` and `IEditorGroup#isActive(...)`

Comment (alexr00) [reply src/vs/workbench/api/browser/mainThreadEditorTabs.ts]: Thanks, I didn't notice that.

File: src/vs/workbench/api/browser/mainThreadEditorTabs.ts [hunk 2]
```diff
@@ -69,7 +72,8 @@ export class MainThreadEditorTabs {
 				tabs.push({
 					group: group.id,
 					name: editor.getTitle(Verbosity.SHORT) ?? '',
-					resource: editor.resource
+					resource: editor.resource,
+					isActive: group.isActive(editor)
```
Comment (jrieken) [inline src/vs/workbench/api/browser/mainThreadEditorTabs.ts:76]: Unsure if this is "active per group" and hence many active editors or just one active editor. Something to test.

**Review by jrieken — APPROVED** [review]
lgtm

**Pull request merged**